pkg/util/azureclient/resources_addons.go

```diff
@@ -6,6 +6,18 @@ import (
 	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
 )
 
+type DeploymentsClientAddons interface {
+	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, deploymentName string, parameters resources.Deployment) error
```

Comment:
this doesn't fit the pattern - it should be called CreateOrUpdate and should be identical to the nested CreateOrUpdate that it calls but for not returning a future

Reply:
agree, but have a look at the vmimage usage. it ssh's into the vm, so it can't wait.